frontend/src/components/RisLawPreview.vue

```diff
@@ -24,18 +26,177 @@ withDefaults(
+   *
+   * Event handlers, a11y properties and so on are only created for elements for which an event handler is registered.
+   *
+   * E.g. `@click:akn:article` will fire for every click on a part of the preview that is showing an article.
+   */
+  [key: `click:akn:${string}`]: [
+    {
+      /**
+       * Eid of the element that was clicked.
+       */
+      eid: string
+      /**
+       * GUID of the element that was clicked.
+       */
+      guid?: string
+      /**
+       * The original event.
+       */
+      originalEvent: MouseEvent | KeyboardEvent
+    },
+  ]
+}>()
+
+const attrs = useAttrs()
+
+/**
+ * The element containing the provided content html.
+ */
+const container = ref<HTMLElement | null>()
+
+/**
+ * Makes the given HTMLElement clickable.
+ *
+ * Handles adding the event listeners (for both mouse and keyboard interactions), a11y role and tab index to the element.
+ *
+ * @param element the HTMLElement to make clickable.
+ * @param eventHandler the handler to call when the element is clicked
+ * @param eventListenerOptions additional options for the event listeners.
+ */
+function makeElementClickable(
+  element: HTMLElement,
+  eventHandler: (event: KeyboardEvent | MouseEvent) => void,
+  eventListenerOptions: AddEventListenerOptions,
+) {
+  element.addEventListener(
+    "click",
+    (event: Event) => {
+      if (!(event instanceof MouseEvent)) return
+      eventHandler(event)
+    },
+    eventListenerOptions,
+  )
+
+  element.addEventListener(
+    "keydown",
+    (event: Event) => {
+      if (
+        !(event instanceof KeyboardEvent) ||
+        !(event.key === "Enter" || event.key === " ")
+      )
+        return
+      eventHandler(event)
+    },
+    eventListenerOptions,
+  )
+
+  element.tabIndex = 0
+  element.role = "button"
+}
+
+/**
+ * Wire up event handling for click events whenever the document changes.
+ */
+watch(
+  () => props.content,
+  async (value, oldValue, onCleanup) => {
+    // Need to tick in order to give Vue some time to render the HTML first
+    await nextTick()
+
+    const abortController = new AbortController()
+    onCleanup(() => abortController.abort())
+
+    Object.keys(attrs)
+      .filter((key) => key.startsWith("onClick:akn:"))
+      .map((key) => key.replace("onClick:akn:", ""))
+      .forEach((aknElement) => {
+        container.value
+          ?.querySelectorAll(`.akn-${aknElement}`)
+          ?.forEach((htmlElement) => {
+            if (!(htmlElement instanceof HTMLElement)) return
+
+            const eid = htmlElement.dataset.eid
+            const guid = htmlElement.dataset.guid
+
+            if (!eid) return
+
+            makeElementClickable(
+              htmlElement,
+              (event) => {
+                emit(`click:akn:${aknElement}`, {
+                  eid,
+                  guid,
+                  originalEvent: event,
+                })
+              },
+              {
+                signal: abortController.signal,
```

Comment:
Curious: what does the `AbortController` do here?

Reply:
Once the abort controller aborts the event listener ist removed. (So it basically replaces the stuff one would otherwise need to do with removeEventListener)